x-pack/filebeat/module/crowdstrike/falcon/config/pipeline.js

```diff
@@ -16,22 +23,41 @@ var crowdstrikeFalcon = (function() {
         max_depth: 8
     });
 
-    var dropFields = function(evt) {
+    var dropIfEmpty = function (evt, field) {
+        var value = evt.Get(field);
+        if (!value || value === "") {
+            evt.Delete(field);
+        }
+    }
+
+    var dropFields = function (evt) {
         evt.Delete("message");
         evt.Delete("host.name");
+        dropIfEmpty(evt, "crowdstrike.event.UserIp");
```

Comment:
Don't think we need this.  In the convert  it should be "fail_on_error: false", not "ignore_failure: true", then we don't need the dropIfEmpty

Reply:
Made the change, but are we ok with passing off an empty string in an `ip` field? You can take a look at the difference in the approach here: https://github.com/elastic/beats/pull/20138/commits/a6d288cc20da017ac3f322f7aa4dd98db683f9bb

cc: @tonymeehan